language/move-model/src/model.rs

```diff
@@ -1188,6 +1189,9 @@ pub struct ModuleData {
 
     /// A cache for the modules used by this one.
     used_modules: RefCell<BTreeMap<bool, BTreeSet<ModuleId>>>,
+
+    /// A cache for the modules declared as friends by this one.
+    friend_modules: RefCell<Option<BTreeSet<ModuleId>>>,
```

Comment:
Nit: maybe drop the option for consistency with `used_modules` and other data fields (which use empty collections instead of `None`)?

Reply:
This is an interesting point actually. I think  using `Option` seems to be more consistent with `used_modules`, the reason is following:

In both cases, the `BTreeSet`s are constructed separately and put into the `used_modules` and `friend_modules` as a whole, instead of directly modifying the BTreeSet in-place. The only difference is that:
- for `used_modules`, it uses `map.insert(true/false, pre_constructed_set)` to insert the set,
- for `friend_modules`, it uses `option.get_or_insert_with(pre_constructed_set)` to insert the set.

Viewing it from another perspective, you can imagine that 
`Option<BTreeSet<ModuleId>>` is the same as `BTreeMap<bool, BTreeSet<ModuleId>>` where the key can only be `true`. Existence of `true` in the `BTreeMap` corresponds to the `Some` variant in the option and non-existence of `true` corresponds to the `None` variant.